src/server/search/search_family.cc

```diff
@@ -51,39 +51,39 @@ search::SchemaField::VectorParams ParseVectorParams(CmdArgParser* parser) {
   size_t num_args = parser->Next<size_t>();
 
   for (size_t i = 0; i * 2 < num_args; i++) {
-    if (parser->Check("DIM").ExpectTail(1)) {
+    if (parser->Check("DIM")) {
       params.dim = parser->Next<size_t>();
       continue;
     }
```

Comment:
Isn't this a bug now?

Before, with `ExpectTail(1)` you enter this branch only if `tag DIM` is followed by at least one argument which you parse below via `parser->Next<size_t>()`. 
Now with the changes introduced and based on:

```
    if (cur_i_ + sizeof...(Ts) >= args_.size()) {
      Report(OUT_OF_BOUNDS, cur_i_);
      return std::conditional_t<sizeof...(Ts) == 0, T, std::tuple<T, Ts...>>();
    }
```

You will return `T=size_t` which is a default constructed size_t. Isn't this a bug ? You are expecting to parse one argument, the user did not supply one and yet you succeeded the parsing and `continued`. 

Am I missing something here or?

Reply:
IMO I think we need `ExpectTail` because that way we don't need to handle the error below. Maybe I am missing something :man_shrugging:

Reply:
parser is checked for error after the cycle, also i * 2 < num_args checks the same

Reply:
oh because we explicitly check `if (auto err = parser.Error(); err)` below right. This was the parser's property that Vlad implemented. Makes sense